### ISSUE TITLE
Craft 5 multi-site install error

### ISSUE THREAD

**Issue by nitinl-cogapp** [solspace/craft3-calendar#297]
**Error on running an install from config (i.e. with no DB): `Exception: No primary site exists`**

**Steps to reproduce**

1. Create a multi-site Craft site
2. Add calendar plugin
3. Re-install the site from the generated config with `php craft install/craft`

**Expected behavior**

Successful installation

**Craft & Plugin Info (please complete the following information):**

- Craft Version: 5.2.3
- Calendar Version: 5.0.5
- Calendar Edition: Pro
- Fresh Install or Upgrade: Fresh install

**Additional context**

```
*** failed to install Craft: An error occurred while executing the "craft\migrations\Install migration: No primary site exists
Exception: No primary site exists (/home/runner/work/<projectname>/cms/vendor/craftcms/cms/src/services/Sites.php:525)

#0 /home/runner/work/<projectname>/cms/vendor/solspace/craft-calendar/packages/plugin/src/Services/EventsService.php(395): craft\services\Sites->getPrimarySite()
#1 [internal function]: Solspace\Calendar\Services\EventsService->addSiteHandler()
```

Also, it's not a config issue, as I've ensured none of the issues mentioned [here](https://github.com/craftcms/cms/issues/7657) are present on my setup.

This is running in our CI, and interestingly this works fine on:
- Craft Version: 4.9.7
- Calendar Version: 5.0.5
- Calendar Edition: Pro
- Fresh Install or Upgrade: Fresh install

**Comment by qrazi** [comment]
A fresh install also fails on 4.10.2 & on 4.10.0 whilst working on 4.9.7 for me as well. I'm still looking into it a bit more and might end up opening a separate issue, but on the surface seems to be at least related.

**Comment by qrazi** [comment]
OK, it is indeed the _same_ issue. Craft 4.10.0 and Craft 5.2.0 introduced slightly different logic for the installer. It added a validation for the project config before actually running the project config. This validation was initially done - but validating less - inline in `\craft\migrations\Install::insertDefaultData()`, the 5th step of `\craft\migrations\Install::safeUp()`.

Since 4.10.0 / 5.2.0 the validation is done as the very first thing in `safeUp()` and it was expended to check that plugins installed with composer are the same as defined in project config. To check this, it runs ` $pluginsService->createPlugin($handle)`, which obviously calls `\Solspace\Calendar\Calendar::init()`, and thus registers all listeners as well.

So what (summarized) happened in <5.2.0 / <4.10.0:

1. `insertDefaultData` runs `ProjectConfigHelper::ensureAllSitesProcessed(true);`
2. This triggers `ProjectConfigService::EVENT_ADD_ITEM` event
3. This triggers a `Sites::EVENT_AFTER_SAVE_SITE` event
4. No listeners are found for `Sites::EVENT_AFTER_SAVE_SITE`

And what happens in >=5.2.0 / >=4.10.0:

1. Project config validation triggers `\Solspace\Calendar\Calendar::init()`, which means a listener is registered for `Sites::EVENT_AFTER_SAVE_SITE`
2. Some steps later `insertDefaultData` runs `ProjectConfigHelper::ensureAllSitesProcessed(true);`
3. This triggers `ProjectConfigService::EVENT_ADD_ITEM` event
4. This triggers a `Sites::EVENT_AFTER_SAVE_SITE` event
5. A listener is found for `Sites::EVENT_AFTER_SAVE_SITE`; the Calendar plugin registered two of them
6. `\Solspace\Calendar\Services\EventsService::addSiteHandler()` is executed - since this is a fresh install, it is also a new Site, and the whole method is executed
7. But, crucially, the plugin migrations haven't run yet at this point, so the queries executed by `\Solspace\Calendar\Services\EventsService::addSiteHandler()`  & `\Solspace\Calendar\Services\CalendarsService::addSiteHandler()` throw an error over non-existing tables

EDIT: looking at the Craft 5 code flow for this method and the error posted by OP, I am not 100% sure if it is indeed the same error. I only tested on 4.10.2 vs 4.9.7 and for me the Installer on 4.10.2 throws an error about missing `calendar_events` table, in the `\Solspace\Calendar\Services\EventsService::addSiteHandler()`-method. Still; most like this change in behavior also causes the error of OP.

**Comment by kjmartens** [comment]
Thanks for reporting this @nitinl-cogapp and @qrazi (and thanks for the PR). We'll investigate it shortly. 🙂

**Comment by qrazi** [comment]
I think I might have been a bit rambly in my comment explaining the cause of the issue. So feel free to ask me to clarify stuff 😅

**Comment by boboldehampsink** [comment]
This is also failing for me on Craft 4

**Comment by nitinl-cogapp** [comment]
Is this fix due to be released soon? I couldn't find any docs for a release schedule you may have. Thanks.

**Comment by kjmartens** [comment]
Sorry for the delay... we should have a new version available later today. 🙂

**Comment by kjmartens** [comment]
This should now be resolved in Calendar 5.0.6+. 🙂
